### PR TITLE
Reg 75 end event message handling replace start event with end event after start

### PR DESCRIPTION
**Pull request by MarekBrandt**
This time gif should work. Please take a look :D

Running Devices - regatta_buddy 2023-11-18 10-58-34

### CODE REVIEW

File: lib/models/message.dart [hunk 1]
```diff
@@ -3,32 +3,27 @@ class Message {
...
-        teamId = json['teamId'],
-        value = json['value'];
+        timestamp = json['timestamp']!;
 
   Map<String, String> toJson() {
     Map<String, String> json = {
       'type': type.name,
-      'receiverType': receiverType.name
+      'receiverType': receiverType.name,
+      'timestamp': timestamp
```
Comment (bartoz1) [inline lib/models/message.dart:26]: do we really need this? the timestamp is already a key

File: lib/services/event_message_handler.dart [hunk 2]
```diff
@@ -46,21 +53,4 @@ class EventMessageHandler {
   void stop() {
     messageStream?.cancel();
   }
-
-  static Future<List<Message>> getAllMessages(String eventId) async {
```
Comment (bartoz1) [inline lib/services/event_message_handler.dart:50]: very good

**Review by jan-Kulpas — APPROVED** [review]

File: lib/services/event_message_sender.dart [hunk 3]
```diff
@@ -10,8 +10,22 @@ class EventMessageSender {
...
         type: MessageType.startEvent,
-        receiverType: MessageReceiverType.all
-    );
+        receiverType: MessageReceiverType.all,
+        timestamp: timestamp);
+
+    newMessage.set(message.toJson());
+  }
+
+  static void endEvent(String eventId) {
```
Comment (bartoz1) [inline lib/services/event_message_sender.dart:19]: maybe adding a verb here would be more readable idk, sendEndEventMessage

File: lib/models/message.dart (hunk 1 above)
Comment (MarekBrandt) [reply lib/models/message.dart:26]: You've added this in you pr. But I would argue that it's better to have it also inside payload, it's easier to get it

File: lib/services/event_message_sender.dart (hunk 3 above)
Comment (MarekBrandt) [reply lib/services/event_message_sender.dart:19]: end is a verb

File: lib/models/message.dart (hunk 1 above)
Comment (bartoz1) [reply lib/models/message.dart:26]: yes but i was not sending this to db as an additional field in the object

**Pull request merged**